modules/core/src/matrix.cpp

```diff
@@ -5,6 +5,8 @@
 #include "precomp.hpp"
 #include "bufferpool.impl.hpp"
 
+#include <mutex>
```

Comment:
This should not be used without guards like this:
- https://github.com/opencv/opencv/blob/4.5.5/modules/core/include/opencv2/core/utility.hpp#L61-L63
- and others like `OPENCV_DISABLE_THREAD_SUPPORT`

Reply:
If we add a guard, what would be in the else case?